File: cases/taint/scala/expression-negative/ExpressionNegative.scala
package dataflowbench

object ExpressionNegative {
  def dfb_source(): Int = { // DFB-SOURCE: expression-negative-input
    1
  }

  def dfb_sink(value: Int): Unit = {} // DFB-SINK: expression-negative-sink

  def run(): Unit = {
    val value = dfb_source()
    val computed = (value * 3) + 7 // DFB-WITNESS: expression-negative-computed
    dfb_sink(7)
  }
}

File: cases/taint/php/expression-negative/expression_negative.php
<?php
function dfb_source(): int { // DFB-SOURCE: expression-negative-input
    return 4;
}

function dfb_sink(int $value): void {} // DFB-SINK: expression-negative-sink

function run(): void {
    $value = dfb_source();
    $computed = ($value * 3) + 7; // DFB-WITNESS: expression-negative-computed
    dfb_sink(7);
}
